modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/FullStateTransferIndexChooser.java

```diff
@@ -317,4 +337,23 @@ private long findStoppingActivationTsBusy(int indexId, int toCatalogVersionInclu
     private Set<Integer> tableIds(int catalogVersion) {
         return catalogService.tables(catalogVersion).stream().map(CatalogObjectDescriptor::id).collect(toSet());
     }
+
+    private int tableVersionBusy(CatalogIndexDescriptor index, int catalogVersion) {
+        CatalogTableDescriptor table = catalogService.table(index.tableId(), catalogVersion);
+
+        assert table != null : "indexId=" + index.id() + ", tableId=" + index.tableId() + ", catalogVersion=" + catalogVersion;
+
+        return table.tableVersion();
+    }
+
+    private List<IndexIdAndTableVersion> toIndexIdAndTableVersionList(List<Integer> indexIds) {
```

Comment:
How about calling this `enrichWithTableVersions`?

Reply:
why not